fix missing junit

garfieldnate opened the pull request:
Builds were failing because `junit` was not installed. Use a dedicated action plugin to ensure it is available and also publish the test report on PR's for convenient viewing.

garfieldnate committed:
fix missing junit

Builds were failing because `junit` was not installed. Use a dedicated
action plugin to ensure it is available and also publish the test report
on PR's for convenient viewing. For now we don't create an actual PR
check, only annotations, because the check is currently sent to the
wrong build. See
https://github.com/mikepenz/action-junit-report/issues/40.

garfieldnate commented:
Opening this one as a PR so we can see what the reporting looks like.

scijones reviewed (approved):
seems to work. 👍

scijones commented:
To be clear on what I mean by "seems to work" -- "seems to be failing for the right reasons".

garfieldnate committed:
Disable ignored failures in XUnit output

A `status="ignored"` with a failure child element would have been
better, this isn't currently supported by action-unit-report. See
https://github.com/mikepenz/action-junit-report/issues/761. Disabling
the test case in the output prevents the build from failing.

garfieldnate commented:
First green build!
